Implemented testrail integration

harshshekhar15 opened the pull request:
**What this PR does / why we need it**:

- Included testrail update block in test-node-eviction-out-of-disk.
- Included deployment of cassandra in its own namespace.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

ksatchit commented:
@harshshekhar15 , can you please attach the cassandra test outputs with changes

harshshekhar15 commented:
```
PLAYBOOK: k8s-cassandra-pod.yml ****************************************************************************************************************
1 plays in playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml

PLAY [localhost] *******************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:16
ok: [localhost]
META: ran handlers

TASK [Get $HOME of K8s master for kubernetes user] *********************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-prereq.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; echo $HOME", "delta": "0:00:00.002538", "end": "2018-08-06 14:16:27.931478", "rc": 0, "start": "2018-08-06 14:16:27.928940", "stderr": "", "stderr_lines": [], "stdout": "/home/harsh", "stdout_lines": ["/home/harsh"]}

TASK [include_tasks] ***************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-prereq.yml:10
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/stern_task.yml for localhost

TASK [Start the log aggregator to capture test pod logs] ***************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/stern_task.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; nohup stern --all-namespaces \"maya*|openebs*|pvc*|cassandra*\" --kubeconfig /home/harsh/.kube/config --since 1m > /home/harsh/setup/logs/hyperconverged_cassandra_db_on_k8s.log &", "delta": "0:00:00.003266", "end": "2018-08-06 14:16:28.161842", "rc": 0, "start": "2018-08-06 14:16:28.158576", "stderr": "mesg: ttyname failed: Inappropriate ioctl for device\n/bin/bash: /home/harsh/setup/logs/hyperconverged_cassandra_db_on_k8s.log: No such file or directory", "stderr_lines": ["mesg: ttyname failed: Inappropriate ioctl for device", "/bin/bash: /home/harsh/setup/logs/hyperconverged_cassandra_db_on_k8s.log: No such file or directory"], "stdout": "", "stdout_lines": []}

TASK [Terminate the log aggregator] ************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/stern_task.yml:14
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [1a) Download YAML for cassandra service] *************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:26
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/get_url.yml for localhost

TASK [Download files] **************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/get_url.yml:1
ok: [localhost -> None] => {"attempts": 1, "changed": false, "checksum_dest": "db47a391a18031183d8a7a572fb21b2ccf7fc2cb", "checksum_src": "db47a391a18031183d8a7a572fb21b2ccf7fc2cb", "dest": "/home/harsh/cassandra-service.yaml", "gid": 1000, "group": "harsh", "md5sum": "70b93b4ddb7526c9a6de6a7d8d836b2a", "mode": "0664", "msg": "OK (165 bytes)", "owner": "harsh", "size": 165, "src": "/tmp/tmpcSYFnW", "state": "file", "status_code": 200, "uid": 1000, "url": "https://raw.githubusercontent.com/openebs/openebs/master/k8s/demo/cassandra/cassandra-service.yaml"}

TASK [2a) Download YAML for cassandra statefulset] *********************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:33
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/get_url.yml for localhost

TASK [Download files] **************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/get_url.yml:1
changed: [localhost -> None] => {"attempts": 1, "changed": true, "checksum_dest": "628030e859a381c8e7a158a609feeeb2c4a53d08", "checksum_src": "92e49a3a0db4d1eb28e8cda44dcb8a6db128b099", "dest": "/home/harsh/cassandra-statefulset.yaml", "gid": 1000, "group": "harsh", "md5sum": "d6474603cbd53093be05318333c58b37", "mode": "0664", "msg": "OK (2382 bytes)", "owner": "harsh", "size": 2382, "src": "/tmp/tmpYJpZj_", "state": "file", "status_code": 200, "uid": 1000, "url": "https://raw.githubusercontent.com/openebs/openebs/master/k8s/demo/cassandra/cassandra-statefulset.yaml"}

TASK [2b) Download YAML for cassandra loadgen] *************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:40
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/get_url.yml for localhost

TASK [Download files] **************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/get_url.yml:1
ok: [localhost -> None] => {"attempts": 1, "changed": false, "checksum_dest": "8ab2292cde2fbcc4258fba90066da5c053f4b14d", "checksum_src": "8ab2292cde2fbcc4258fba90066da5c053f4b14d", "dest": "/home/harsh/cassandra-loadgen.yaml", "gid": 1000, "group": "harsh", "md5sum": "a3522bc82883b90ad5b528e9efaf62c5", "mode": "0664", "msg": "OK (393 bytes)", "owner": "harsh", "size": 393, "src": "/tmp/tmp8aVt8B", "state": "file", "status_code": 200, "uid": 1000, "url": "https://raw.githubusercontent.com/openebs/openebs/master/k8s/demo/cassandra/cassandra-loadgen.yaml"}

TASK [2c) Replace the default namespace in cassandra statefulset yaml] *************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:47
changed: [localhost] => {"changed": true, "msg": "1 replacements made"}

TASK [Creating namespace for cassandra] ********************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:54
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl create ns cassandra", "delta": "0:00:01.102719", "end": "2018-08-06 14:16:31.372444", "rc": 0, "start": "2018-08-06 14:16:30.269725", "stderr": "", "stderr_lines": [], "stdout": "namespace/cassandra created", "stdout_lines": ["namespace/cassandra created"]}

TASK [3) Get the number of nodes in the cluster] ***********************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:60
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get nodes | grep 'Ready' | grep 'none' | wc -l", "delta": "0:00:01.723779", "end": "2018-08-06 14:16:33.284572", "rc": 0, "start": "2018-08-06 14:16:31.560793", "stderr": "", "stderr_lines": [], "stdout": "3", "stdout_lines": ["3"]}

TASK [3a) Fetch the node count from stdout] ****************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:67
ok: [localhost] => {"ansible_facts": {"node_count": " 3"}, "changed": false}

TASK [3b) Replace the replica count in cassandra statefulset yaml] *****************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:71
changed: [localhost -> None] => {"changed": true, "msg": "1 replacements made"}

TASK [3c) Replace the io_duration in cassandra loadgen yaml] ***********************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:78
changed: [localhost -> None] => {"changed": true, "msg": "1 replacements made"}

TASK [Check whether maya-apiserver pod is deployed] ********************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:85
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check maya-api pod status] ***************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n openebs | grep maya-api", "delta": "0:00:01.307008", "end": "2018-08-06 14:16:35.178835", "rc": 0, "start": "2018-08-06 14:16:33.871827", "stderr": "", "stderr_lines": [], "stdout": "maya-apiserver-dc8f6bf4d-qvjm5             1/1       Running   0          2d", "stdout_lines": ["maya-apiserver-dc8f6bf4d-qvjm5             1/1       Running   0          2d"]}

TASK [4) Deploy cassandra service] *************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:91
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_task.yml for localhost

TASK [Deploy application files] ****************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_task.yml:2
changed: [localhost -> None] => (item=cassandra-service.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f \"/home/harsh/cassandra-service.yaml\" -n cassandra", "delta": "0:00:02.738232", "end": "2018-08-06 14:16:38.156227", "item": "cassandra-service.yaml", "rc": 0, "start": "2018-08-06 14:16:35.417995", "stderr": "", "stderr_lines": [], "stdout": "service/cassandra created", "stdout_lines": ["service/cassandra created"]}

TASK [4a) Deploy cassandra statefulset] ********************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:97
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_task.yml for localhost

TASK [Deploy application files] ****************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_task.yml:2
changed: [localhost -> None] => (item=cassandra-statefulset.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f \"/home/harsh/cassandra-statefulset.yaml\" -n cassandra", "delta": "0:00:03.091184", "end": "2018-08-06 14:16:41.505593", "item": "cassandra-statefulset.yaml", "rc": 0, "start": "2018-08-06 14:16:38.414409", "stderr": "", "stderr_lines": [], "stdout": "statefulset.apps/cassandra created", "stdout_lines": ["statefulset.apps/cassandra created"]}

TASK [5) Confirm cassandra pod status is running] **********************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:103
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check cassandra-0 pod status] ************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
FAILED - RETRYING: Check cassandra-0 pod status (15 retries left).
FAILED - RETRYING: Check cassandra-0 pod status (14 retries left).
changed: [localhost -> None] => {"attempts": 3, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n cassandra | grep cassandra-0", "delta": "0:00:01.416609", "end": "2018-08-06 14:17:46.624287", "rc": 0, "start": "2018-08-06 14:17:45.207678", "stderr": "", "stderr_lines": [], "stdout": "cassandra-0                                                      0/1       Running   0          1m", "stdout_lines": ["cassandra-0                                                      0/1       Running   0          1m"]}

TASK [6) Start cassandra load generation] ******************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:109
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_task.yml for localhost

TASK [Deploy application files] ****************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_task.yml:2
changed: [localhost -> None] => (item=cassandra-loadgen.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f \"/home/harsh/cassandra-loadgen.yaml\" -n cassandra", "delta": "0:00:03.088593", "end": "2018-08-06 14:17:49.945934", "item": "cassandra-loadgen.yaml", "rc": 0, "start": "2018-08-06 14:17:46.857341", "stderr": "", "stderr_lines": [], "stdout": "job.batch/cassandra-loadgen created", "stdout_lines": ["job.batch/cassandra-loadgen created"]}

TASK [6a) Verify load is running for specified duration] ***************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:115
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check cassandra-loadgen pod status] ******************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
FAILED - RETRYING: Check cassandra-loadgen pod status (15 retries left).
changed: [localhost -> None] => {"attempts": 2, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n cassandra | grep cassandra-loadgen", "delta": "0:00:01.572290", "end": "2018-08-06 14:18:23.399776", "rc": 0, "start": "2018-08-06 14:18:21.827486", "stderr": "", "stderr_lines": [], "stdout": "cassandra-loadgen-2ww6c                                          1/1       Running             0          34s", "stdout_lines": ["cassandra-loadgen-2ww6c                                          1/1       Running             0          34s"]}

TASK [Wait for 180 secs to run load.] **********************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:121
ok: [localhost] => {"changed": false, "elapsed": 180, "path": null, "port": null, "search_regex": null, "state": "started"}

TASK [6b) Verify load by using describe keyspaces] *********************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:125
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl exec cassandra-0 -n cassandra -- cqlsh --execute \"describe keyspaces;\"", "delta": "0:00:05.218596", "end": "2018-08-06 14:21:29.234666", "rc": 0, "start": "2018-08-06 14:21:24.016070", "stderr": "", "stderr_lines": [], "stdout": "\nsystem_schema  system_auth  system  keyspace1  system_distributed  system_traces", "stdout_lines": ["", "system_schema  system_auth  system  keyspace1  system_distributed  system_traces"]}

TASK [Test Passed] *****************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:138
ok: [localhost] => {"ansible_facts": {"flag": "Test Passed", "status_id": 1}, "changed": false}

TASK [Set Status] ******************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:143
ok: [localhost] => {"ansible_facts": {"status": "good"}, "changed": false}

TASK [Delete cassandra service pod] ************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-cleanup.yml:2
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml for localhost

TASK [Delete application] **********************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml:2
changed: [localhost -> None] => (item=cassandra-service.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f \"/home/harsh/cassandra-service.yaml\" -n cassandra", "delta": "0:00:01.519643", "end": "2018-08-06 14:21:31.118777", "item": "cassandra-service.yaml", "rc": 0, "start": "2018-08-06 14:21:29.599134", "stderr": "", "stderr_lines": [], "stdout": "service \"cassandra\" deleted", "stdout_lines": ["service \"cassandra\" deleted"]}

TASK [Delete cassandra statefulset pod] ********************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-cleanup.yml:8
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml for localhost

TASK [Delete application] **********************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml:2
changed: [localhost -> None] => (item=cassandra-statefulset.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f \"/home/harsh/cassandra-statefulset.yaml\" -n cassandra", "delta": "0:00:02.108461", "end": "2018-08-06 14:21:33.532766", "item": "cassandra-statefulset.yaml", "rc": 0, "start": "2018-08-06 14:21:31.424305", "stderr": "", "stderr_lines": [], "stdout": "statefulset.apps \"cassandra\" deleted", "stdout_lines": ["statefulset.apps \"cassandra\" deleted"]}

TASK [Delete cassandra loadgen pod] ************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-cleanup.yml:14
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml for localhost

TASK [Delete application] **********************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml:2
changed: [localhost -> None] => (item=cassandra-loadgen.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f \"/home/harsh/cassandra-loadgen.yaml\" -n cassandra", "delta": "0:00:01.795157", "end": "2018-08-06 14:21:35.604283", "item": "cassandra-loadgen.yaml", "rc": 0, "start": "2018-08-06 14:21:33.809126", "stderr": "", "stderr_lines": [], "stdout": "job.batch \"cassandra-loadgen\" deleted", "stdout_lines": ["job.batch \"cassandra-loadgen\" deleted"]}

TASK [Get cassandra pvcs] **********************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-cleanup.yml:20
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get pvc -n cassandra -o custom-columns=:metadata.name --no-headers | grep cassandra-data", "delta": "0:00:02.935035", "end": "2018-08-06 14:21:38.735108", "rc": 0, "start": "2018-08-06 14:21:35.800073", "stderr": "", "stderr_lines": [], "stdout": "cassandra-data-cassandra-0\ncassandra-data-cassandra-1\ncassandra-data-cassandra-2", "stdout_lines": ["cassandra-data-cassandra-0", "cassandra-data-cassandra-1", "cassandra-data-cassandra-2"]}

TASK [Delete cassandra pvcs] *******************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-cleanup.yml:28
changed: [localhost -> None] => (item=cassandra-data-cassandra-0) => {"changed": true, "cmd": "source ~/.profile; kubectl delete pvc cassandra-data-cassandra-0 -n cassandra", "delta": "0:00:01.597008", "end": "2018-08-06 14:21:40.535659", "item": "cassandra-data-cassandra-0", "rc": 0, "start": "2018-08-06 14:21:38.938651", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim \"cassandra-data-cassandra-0\" deleted", "stdout_lines": ["persistentvolumeclaim \"cassandra-data-cassandra-0\" deleted"]}
changed: [localhost -> None] => (item=cassandra-data-cassandra-1) => {"changed": true, "cmd": "source ~/.profile; kubectl delete pvc cassandra-data-cassandra-1 -n cassandra", "delta": "0:00:02.365334", "end": "2018-08-06 14:21:43.059569", "item": "cassandra-data-cassandra-1", "rc": 0, "start": "2018-08-06 14:21:40.694235", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim \"cassandra-data-cassandra-1\" deleted", "stdout_lines": ["persistentvolumeclaim \"cassandra-data-cassandra-1\" deleted"]}
changed: [localhost -> None] => (item=cassandra-data-cassandra-2) => {"changed": true, "cmd": "source ~/.profile; kubectl delete pvc cassandra-data-cassandra-2 -n cassandra", "delta": "0:00:01.393963", "end": "2018-08-06 14:21:44.629742", "item": "cassandra-data-cassandra-2", "rc": 0, "start": "2018-08-06 14:21:43.235779", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim \"cassandra-data-cassandra-2\" deleted", "stdout_lines": ["persistentvolumeclaim \"cassandra-data-cassandra-2\" deleted"]}

TASK [Confirm cassandra pod has been deleted] **************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-cleanup.yml:37
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods", "delta": "0:00:01.065695", "end": "2018-08-06 14:21:45.893112", "rc": 0, "start": "2018-08-06 14:21:44.827417", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}

TASK [Get cassandra deploy] ********************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-cleanup.yml:48
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get deploy -n cassandra --no-headers | awk {'print $1'}", "delta": "0:00:02.068956", "end": "2018-08-06 14:21:48.153271", "rc": 0, "start": "2018-08-06 14:21:46.084315", "stderr": "", "stderr_lines": [], "stdout": "pvc-6b83db48-9955-11e8-97c9-42010a80002c-rep\npvc-9b89b8d3-9955-11e8-97c9-42010a80002c-ctrl\npvc-9b89b8d3-9955-11e8-97c9-42010a80002c-rep", "stdout_lines": ["pvc-6b83db48-9955-11e8-97c9-42010a80002c-rep", "pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-ctrl", "pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-rep"]}

TASK [Delete cassandra deploy] *****************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-cleanup.yml:56
changed: [localhost -> None] => (item=pvc-6b83db48-9955-11e8-97c9-42010a80002c-rep) => {"changed": true, "cmd": "source ~/.profile; kubectl delete deploy pvc-6b83db48-9955-11e8-97c9-42010a80002c-rep -n cassandra", "delta": "0:00:01.754868", "end": "2018-08-06 14:21:50.101092", "item": "pvc-6b83db48-9955-11e8-97c9-42010a80002c-rep", "rc": 0, "start": "2018-08-06 14:21:48.346224", "stderr": "", "stderr_lines": [], "stdout": "deployment.extensions \"pvc-6b83db48-9955-11e8-97c9-42010a80002c-rep\" deleted", "stdout_lines": ["deployment.extensions \"pvc-6b83db48-9955-11e8-97c9-42010a80002c-rep\" deleted"]}
changed: [localhost -> None] => (item=pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-ctrl) => {"changed": true, "cmd": "source ~/.profile; kubectl delete deploy pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-ctrl -n cassandra", "delta": "0:00:01.605941", "end": "2018-08-06 14:21:51.865474", "item": "pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-ctrl", "rc": 0, "start": "2018-08-06 14:21:50.259533", "stderr": "", "stderr_lines": [], "stdout": "deployment.extensions \"pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-ctrl\" deleted", "stdout_lines": ["deployment.extensions \"pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-ctrl\" deleted"]}
changed: [localhost -> None] => (item=pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-rep) => {"changed": true, "cmd": "source ~/.profile; kubectl delete deploy pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-rep -n cassandra", "delta": "0:00:01.628864", "end": "2018-08-06 14:21:53.656250", "item": "pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-rep", "rc": 0, "start": "2018-08-06 14:21:52.027386", "stderr": "", "stderr_lines": [], "stdout": "deployment.extensions \"pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-rep\" deleted", "stdout_lines": ["deployment.extensions \"pvc-9b89b8d3-9955-11e8-97c9-42010a80002c-rep\" deleted"]}

TASK [Delete namespace] ************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-cleanup.yml:66
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl delete ns cassandra", "delta": "0:00:23.396265", "end": "2018-08-06 14:22:17.258703", "rc": 0, "start": "2018-08-06 14:21:53.862438", "stderr": "", "stderr_lines": [], "stdout": "namespace \"cassandra\" deleted", "stdout_lines": ["namespace \"cassandra\" deleted"]}

TASK [Remove test artifacts] *******************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod-cleanup.yml:77
changed: [localhost -> None] => (item=cassandra-service.yaml) => {"changed": true, "item": "cassandra-service.yaml", "path": "/home/harsh/cassandra-service.yaml", "state": "absent"}
changed: [localhost -> None] => (item=cassandra-statefulset.yaml) => {"changed": true, "item": "cassandra-statefulset.yaml", "path": "/home/harsh/cassandra-statefulset.yaml", "state": "absent"}
changed: [localhost -> None] => (item=cassandra-loadgen.yaml) => {"changed": true, "item": "cassandra-loadgen.yaml", "path": "/home/harsh/cassandra-loadgen.yaml", "state": "absent"}

TASK [Test Cleanup Passed] *********************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:163
ok: [localhost] => {"ansible_facts": {"cflag": "Cleanup Passed"}, "changed": false}

TASK [include_tasks] ***************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:174
included: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/stern_task.yml for localhost

TASK [Start the log aggregator to capture test pod logs] ***************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/stern_task.yml:2
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Terminate the log aggregator] ************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/utils/stern_task.yml:14
fatal: [localhost -> None]: FAILED! => {"changed": true, "cmd": "source ~/.profile; killall stern", "delta": "0:00:00.007003", "end": "2018-08-06 14:22:18.297422", "msg": "non-zero return code", "rc": 1, "start": "2018-08-06 14:22:18.290419", "stderr": "mesg: ttyname failed: Inappropriate ioctl for device\nstern: no process found", "stderr_lines": ["mesg: ttyname failed: Inappropriate ioctl for device", "stern: no process found"], "stdout": "", "stdout_lines": []}
...ignoring

TASK [Update Testrail] *************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:179
changed: [localhost] => {"changed": true, "cmd": "source ~/.profile; python3 /home/harsh/case/openebs/e2e/ansible/files/updater.py -tuser karthik.s@cloudbyte.com -tpass openebs -sid 826 -cid 166362 -stid 1", "delta": "0:00:00.085827", "end": "2018-08-06 14:22:18.551365", "rc": 0, "start": "2018-08-06 14:22:18.465538", "stderr": "", "stderr_lines": [], "stdout": "File operation failed, Error: [Errno 2] No such file or directory: '/home/harsh/mapping.json'\nerror - [Errno 2] No such file or directory: '/home/harsh/mapping.json'", "stdout_lines": ["File operation failed, Error: [Errno 2] No such file or directory: '/home/harsh/mapping.json'", "error - [Errno 2] No such file or directory: '/home/harsh/mapping.json'"]}

TASK [Testrail Updated] ************************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:184
ok: [localhost] => {"ansible_facts": {"tflag": "Testrail Updated"}, "changed": false}

TASK [Send slack notification] *****************************************************************************************************************
task path: /home/harsh/case/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-cassandra-pod/k8s-cassandra-pod.yml:193
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Slack has updated its webhook API.  You need to specify a token of the form XXXX/YYYY/ZZZZ in your playbook"}

PLAY RECAP *************************************************************************************************************************************
localhost                  : ok=50   changed=27   unreachable=0    failed=1   
```
